shared-module/msgpack/__init__.c

```diff
@@ -195,19 +196,34 @@ STATIC mp_map_elem_t *dict_iter_next(mp_obj_dict_t *dict, size_t *cur) {
     return NULL;
 }
 
-STATIC void pack_int(msgpack_stream_t *s, int32_t x) {
-    if (x > -32 && x < 128) {
-        write1(s, x);
-    } else if ((int8_t)x == x) {
-        write1(s, 0xd0);
-        write1(s, x);
-    } else if ((int16_t)x == x) {
-        write1(s, 0xd1);
-        write2(s, x);
+STATIC void pack_int(msgpack_stream_t *s, mp_obj_t obj, bool _signed) {
+    byte buffer[9];
```

Comment:
This module is already setup with `write` functions to write its output. It seems unnecessary to add another layer of indirection (`buffer`) to write the output.